### PR TITLE
Hide player version choices if not needed in selected testsuite

### PR DESCRIPTION
**Pull request by lukasz-rutkowski-red**

### CODE REVIEW

File: patches/0001_js_mse_eme_mvt.patch [hunk 1]
```diff
@@ -402,28 +404,35 @@ index c040757..bf5192a 100644
...
      }
      for (var i = 0; i < selectors.length; ++i) {
-       switchDiv.appendChild(document.createTextNode(selectors[i].text));
+-      switchDiv.appendChild(document.createTextNode(selectors[i].text));
 -      var select = document.createElement('select');
-+      var select = document.createElement('span');
-+      select.id = selectors[i].id;
-       for (var j = 0; j < selectors[i].optionTexts.length; ++j) {
+-      for (var j = 0; j < selectors[i].optionTexts.length; ++j) {
 -        select.appendChild(createOption(selectors[i].optionTexts[j],
```
Comment (maciej-kolanski-red) [inline patches/0001_js_mse_eme_mvt.patch:414]: createOption is now unused, please remove our related changes from .patch file. Do not remove whole function, just what was added by us (in order to minimize the patch file):
```
-var createOption = function(text, value) {
+var createOption = function(text, value, isSelected = false) {
   var option = document.createElement('option');
   option.text = text;
   option.value = value;
+  isSelected && (option.selected = 'selected');
   return option;
 };
```

File: patches/0001_js_mse_eme_mvt.patch [hunk 2]
```diff
@@ -402,28 +404,35 @@ index c040757..bf5192a 100644
...
-+        if (selectors[i].optionTexts[j] != selectors[i].defaultValue)
-+            select.lastChild.classList.add('focusable');
-+        else
-+            select.lastChild.classList.add('bold');
++      if (harnessConfig.testType.includes(selectors[i].id)) {
++        switchDiv.appendChild(document.createTextNode(selectors[i].text));
++        var select = document.createElement('span');
++        select.id = selectors[i].id;
++        for (var j = 0; j < selectors[i].optionTexts.length; ++j) {
++          select.appendChild(createAnchor(selectors[i].optionTexts[j],selectors[i].optionTexts[j]));
```
Comment (maciej-kolanski-red) [inline patches/0001_js_mse_eme_mvt.patch:421]: Missing space between arguments

File: patches/0001_js_mse_eme_mvt.patch [hunk 3]
```diff
@@ -402,28 +404,35 @@ index c040757..bf5192a 100644
...
++          select.lastChild.setAttribute('data-href', location.origin + "?test_type=" + harnessConfig.testType + "&engine_" + selectors[i].id + "=" + selectors[i].optionTexts[j]);
```
Comment (maciej-kolanski-red) [inline patches/0001_js_mse_eme_mvt.patch:422]: This line is far too long, please split it somehow. You can use a temporary variable for the url:
```
          let url = location.origin;+ "?test_type=" + harnessConfig.testType;
          url += "&engine_" + selectors[i].id + "=" + selectors[i].optionTexts[j];
          select.lastChild.setAttribute('data-href', url);
```

**Review by maciej-kolanski-red — CHANGES_REQUESTED** [review]

File: patches/0001_js_mse_eme_mvt.patch (hunk 2 above)
Comment (lukasz-rutkowski-red) [reply patches/0001_js_mse_eme_mvt.patch]: Fixed.

**Pull request merged**